### PR TITLE
UC01-020: Fix nameres for IteratedAssocs.

### PR DESCRIPTION
**Pull request by Roldak**

### CODE REVIEW

File: ada/ast.py [hunk 1]
```diff
@@ -14719,9 +14725,19 @@ def expr_type():
             Self.id_type.is_null,
 
             # The type of a for loop variable does not need to be annotated, it
-            # can eventually be infered, which necessitates name resolution on
+            # can eventually be inferred, which necessitates name resolution on
             # the loop specification. Run resolution if necessary.
-            Entity.id.expression_type,
+            # NOTE: if we are in the context of an Ada 2020x iterated component
```
Comment (thvnx) [inline ada/ast.py:14730]: 2020x -> 202x

**Review by thvnx — DISMISSED** [review]
LGTM!

File: ada/ast.py (hunk 1 above)
Comment (Roldak) [reply ada/ast.py]: oops fixed!

File: testsuite/tests/name_resolution/iterated_component_association/iterassoc.adb [hunk 2]
```diff
@@ -1,22 +1,32 @@
-package body Iterassoc is
+ procedure Iterassoc is
```
Comment (thvnx) [inline testsuite/tests/name_resolution/iterated_component_association/iterassoc.adb:1]: Looks like there is a whitespace in the first column of each line here.

**Review by thvnx — APPROVED** [review]

**Pull request merged**